src/components/createPost/searchMaps.tsx

```diff
@@ -0,0 +1,181 @@
+  offset: number;
+  length: number;
+}
+interface StructuredFormatting {
+  main_text: string;
+  secondary_text: string;
+  main_text_matched_substrings?: readonly MainTextMatchedSubstrings[];
+}
+interface PlaceType {
+  description: string;
+  structured_formatting: StructuredFormatting;
+}
+
+export default function GoogleMaps(props: { initialValue?: string }) {
+  const [value, setValue] = React.useState<PlaceType | null>(null);
+  const [inputValue, setInputValue] = React.useState(props.initialValue ?? "");
+  const [options, setOptions] = React.useState<readonly PlaceType[]>([]);
+  const loaded = React.useRef(false);
+
+  if (typeof window !== "undefined" && !loaded.current) {
+    if (!document.querySelector("#google-maps")) {
+      loadScript(
+        `https://maps.googleapis.com/maps/api/js?key=${GOOGLE_MAPS_API_KEY}&libraries=places&callback=Function.prototype`,
+        document.querySelector("head"),
+        "google-maps"
+      );
+    }
+
+    loaded.current = true;
+  }
+
+  const fetch = React.useMemo(
+    () =>
+      debounce(
+        (
+          request: { input: string },
+          callback: (results?: readonly PlaceType[]) => void
+        ) => {
+          (autocompleteService.current as any).getPlacePredictions(
+            request,
+            callback
+          );
+        },
+        400
+      ),
+    []
+  );
+
+  React.useEffect(() => {
+    let active = true;
+
+    if (!autocompleteService.current && (window as any).google) {
+      autocompleteService.current = new (
+        window as any
+      ).google.maps.places.AutocompleteService();
+    }
+    if (!autocompleteService.current) {
+      return undefined;
+    }
+
+    if (inputValue === "") {
+      setOptions(value ? [value] : []);
+      return undefined;
+    }
+
+    fetch({ input: inputValue }, (results?: readonly PlaceType[]) => {
+      if (active) {
+        let newOptions: readonly PlaceType[] = [];
+
+        if (value) {
+          newOptions = [value];
+        }
+
+        if (results) {
+          newOptions = [...newOptions, ...results];
+        }
+
+        setOptions(newOptions);
+      }
+    });
+
+    return () => {
+      active = false;
+    };
+  }, [value, inputValue, fetch]);
+
+  return (
+    <Autocomplete
+      id="google-map-demo"
+      getOptionLabel={(option) =>
+        typeof option === "string" ? option : option.description
+      }
+      filterOptions={(x) => x}
+      options={options}
+      autoComplete
+      includeInputInList
+      filterSelectedOptions
+      value={value}
+      noOptionsText="No locations"
+      onChange={(event: any, newValue: PlaceType | null) => {
+        setOptions(newValue ? [newValue, ...options] : options);
+        setValue(newValue);
+      }}
+      onInputChange={(event, newInputValue) => {
+        setInputValue(newInputValue);
+      }}
+      renderInput={(params) => (
+        <TextField {...params} label="กรอกสถานที่" fullWidth />
```

Comment:
I think the label should not appear when selecting the location like in this picture. I think it should be use place holder instead of label
![image](https://user-images.githubusercontent.com/62459518/218383231-e7c778e7-a301-45e6-8dec-960be0cd70ed.png)